build/vsts-ci.yml

```diff
@@ -31,6 +31,9 @@ variables:
   value: 1es-windows-2019
 - name: MacImage
   value: macOS-12
+- ${{ if and(notin(variables['Build.Reason'], 'PullRequest'), eq(variables['Build.SourceBranch'], 'refs/heads/main')) }}:
```

Comment:
Can you schedule a build on your branch to test this?  You'll need to push your branch to internal.

Reply:
I stole that conditional as-is from aspnetcore

Reply:
I've smoke tested the change. With the branch name changed, https://dev.azure.com/dnceng/internal/_build/results?buildId=2492159 tried to schedule the SourceIndexStage1 job, and https://dev.azure.com/dnceng/internal/_build/results?buildId=2492158 with the old branch name doesn't. So it's doing what it should.